aten/src/ATen/SparseGCSTensorImpl.h

```diff
@@ -0,0 +1,63 @@
+#pragma once
+
+#include <ATen/Tensor.h>
+#include <ATen/TensorUtils.h>
+#include <ATen/SparseTensorUtils.h>
+#include <ATen/SparseCOOTensorImpl.h>
+#include <c10/core/TensorImpl.h>
+#include <c10/util/Exception.h>
+
+namespace at {
+  
+// TODO: since many methods in SparseCOOTensorImpl can be used by GCS sparse tensor directly
+// we probably should have some superclass between TensorImpl and GCSTensorImpl that is
+// shared between COO and GCS tensors.
+struct CAFFE2_API SparseGCSTensorImpl : public TensorImpl {
+  Tensor pointers_;
+  Tensor indices_;
+  Tensor values_;
+  Tensor reduction_;
+
+  // Data for making index conversion operations faster.
+
+  // strides of the first half of the split dimensions.
+  std::vector<int> strides0_;
```

Comment:
Use of `int` here is pretty weird; `int64_t` in our codebase is standard because we often deal with sizes that overflow 32 bit integers.

Reply:
I had to make all the indices use int since pytorch links with LP64 MKL. Could you please have a look at this comment? https://github.com/pytorch/pytorch/pull/44190#issuecomment-747918640

Reply:
For now I will use int32 and put guards to ensure that users do not use 64 bit ints. Linking with ILP64 MKL is crucial for making this 64 bit in the future.

Reply:
Resolving this for now until the MKL issue is sorted.